frontend-ssr/pages/submissions.tsx

```diff
@@ -22,22 +22,49 @@ interface State {
 }
 
 class Submissions extends React.Component<ConnectedProps<typeof connector>, State> {
+  private observer: IntersectionObserver;
+
+  private hiddenRef = React.createRef<HTMLDivElement>();
+
   constructor(props) {
     super(props);
 
     this.state = {
-      loading: props.submissions.submissions === undefined,
+      loading: false,
     };
   }
 
   componentDidMount() {
-    let { submissions } = this.props;
+    this.initIntersectionObserver();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { submissions } = this.props;
 
-    if (submissions.submissions === undefined) {
-      this.getData();
+    if (submissions !== prevProps.submissions && submissions.end) {
```

Comment:
The first part of the `if` is useless. Everytime `submissions` will be different from `prevProps.submissions` due to the nature of Redux which always returns new objects.

You should check whether we reached the end and the previous value was false

`prevProps.submissions.end !== submissions.end && submissions.end`